language/move-ir/types/src/location.rs

```diff
@@ -25,19 +27,19 @@ pub type ByteIndex = u32;
 /// byte vector
 pub struct Loc {
     /// The file the location points to
-    file: &'static str,
+    file: Symbol,
     /// The start byte index into file
     start: ByteIndex,
     /// The end byte index into file
     end: ByteIndex,
 }
 
 impl Loc {
-    pub fn new(file: &'static str, start: ByteIndex, end: ByteIndex) -> Loc {
+    pub fn new(file: Symbol, start: ByteIndex, end: ByteIndex) -> Loc {
         Loc { file, start, end }
     }
 
-    pub fn file(self) -> &'static str {
+    pub fn file(&self) -> Symbol {
```

Comment:
Do we need the ref here? not sure what is more efficient or not 
Maybe we can mark some of these  as always inline?

Reply:
Ah, no, it's not needed. I also don't know which is more efficient, but I'll remove it for now since it's an unnecessary part of this PR.

> Maybe we can mark some of these as always inline?

I'll try profiling `self` vs `&self` and marking this and the other `Loc` methods as inline! In my experience inlining isn't always beneficial as it can increase binary size and result in slower programs. But if these are on especially hot paths and the compiler isn't choosing to inline them for some reason, it could result in some nice speedups. I'll try some of the tips you wrote down the other day.